app/controllers/articles_controller.rb

```diff
@@ -1,7 +1,5 @@
 class ArticlesController < ApplicationController
-
-  http_basic_authenticate_with name: "dhh", password: "secret", except: [:index, :show]
-
+  before_action :validate_session
```

Comment:
@idprietop what a **before_action** means ?

Reply:
before_action means that before executing any action in the controller, what is defined in the before_action will be executed first.